core/store/models/common.go

```diff
@@ -183,32 +183,51 @@ func (j JSON) Bytes() []byte {
 	return []byte(j.String())
 }
 
-// Add returns a new instance of JSON with the new value added.
-func (j JSON) Add(insertKey string, insertValue interface{}) (JSON, error) {
+// AsMap returns j as a map
+func (j JSON) AsMap() (map[string]interface{}, error) {
 	output := make(map[string]interface{})
-
 	switch v := j.Result.Value().(type) {
 	case map[string]interface{}:
 		for key, value := range v {
-			if key != insertKey {
-				output[key] = value
-			}
+			output[key] = value
 		}
-		output[insertKey] = insertValue
 	case nil:
-		output[insertKey] = insertValue
 	default:
-		return JSON{}, errors.New("can only add to JSON objects or null")
+		return nil, errors.New("can only add to JSON objects or null")
 	}
+	return output, nil
+}
 
-	bytes, err := json.Marshal(output)
+// NewJSONFromMap returns m as a JSON object, or errors
+func NewJSONFromMap(m map[string]interface{}) (JSON, error) {
+	bytes, err := json.Marshal(m)
 	if err != nil {
 		return JSON{}, err
 	}
-
 	return JSON{Result: gjson.ParseBytes(bytes)}, nil
 }
```

Comment:
I have code prepared to introduce NewJSON(interface{}), I think this generalized version will take car of this case.